src/updater.tsx

```diff
@@ -99,7 +100,7 @@ export function activeListeningKeys(
 
     memo[callKey] = Object.keys(keyListeners)
       .filter((key) => {
-        const blocksPerFetch = parseInt(key)
+        const blocksPerFetch = !!listenerOptions?.blocksPerFetch ? listenerOptions.blocksPerFetch : parseInt(key)
```

Comment:
I think at this point, the blocksPerFetch should be encoded into the state, so that you do not need to pass `listenerOptions` into this function.

In other words, I think you can revert L91-92, L103, L248-249.

Reply:
You're right this was an important call out actually because I _should_ be getting blocksPerFetch directly from callListeners. What is happening here is wrong. This is because hooks can override the default listenerOptions saved in state, and when they do so by calling addMulticallListeners in slice.ts, the preferred blocksPerFetch is saved in callListeners in state. So activeListeningKeys and outdatedListeningKeys should be getting blocksPerFetch from callListeners in state, not the listenerOptions passed in thru Updater props.